src/main/groovy/com/bmuschko/gradle/clover/InstrumentCodeAction.groovy

```diff
@@ -95,7 +95,13 @@ class InstrumentCodeAction implements Action<Task> {
                 }
 
                 getMethodContexts().each {
-                    ant.methodContext(name: it.name, regexp: it.regexp)
+                    ant.methodContext(
```

Comment:
That was my second choice. Cool, I can do that. The reason i hesitated to
suggest this is that we don't need to use them in any polymorphic context
as far as I know, so they hierarchy only saves us from typing two fields.
Thanks.
On Sat, Jan 17, 2015 at 10:31 AM Benjamin Muschko <notifications@github.com>
wrote:

> In src/main/groovy/com/bmuschko/gradle/clover/InstrumentCodeAction.groovy
> <https://github.com/bmuschko/gradle-clover-plugin/pull/56#discussion-diff-23125966>
> :
>
> > @@ -95,7 +95,13 @@ class InstrumentCodeAction implements Action<Task> {
> >                  }
> >
> >                  getMethodContexts().each {
> > -                    ant.methodContext(name: it.name, regexp: it.regexp)
> > +                    ant.methodContext(
>
> If statement and method contexts have the same fields then you can just
> add them to CloverContextConvention. Otherwise, we'll need to have two
> different classes. In terms of the class hierarchy it depends on what they
> share. If they share name and regexp then you can probably use
> CloverContextConvention as super class for CloverMethodContextConvention.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bmuschko/gradle-clover-plugin/pull/56/files#r23125966>
> .
>